server/request_record.go

```diff
@@ -0,0 +1,49 @@
+package server
+
+import (
+	"github.com/flashbots/rpc-endpoint/database"
+	"github.com/flashbots/rpc-endpoint/types"
+	"github.com/google/uuid"
+	"net/http"
+	"sync"
+)
+
+type requestRecord struct {
+	requestEntry        *database.RequestEntry
+	ethSendRawTxEntries []*database.EthSendRawTxEntry
+	mutex               *sync.Mutex
+}
+
+func NewRequestRecord() *requestRecord {
+	return &requestRecord{
+		requestEntry:        &database.RequestEntry{},
+		ethSendRawTxEntries: make([]*database.EthSendRawTxEntry, 0),
+		mutex:               &sync.Mutex{},
+	}
+}
+
+func (r *requestRecord) AddEthSendRawTxEntry(jsonReq *types.JsonRpcRequest, id, requestId uuid.UUID) *database.EthSendRawTxEntry {
+	var entry *database.EthSendRawTxEntry
+	if jsonReq.Method == "eth_sendRawTransaction" {
```

Comment:
i'm not a fan that this function does or not does it, depending on jsonReq. I think this function should ideally not take jsonReq at all, and maybe the RpcRequest should call this at the correct spot? 🤔

Reply:
Done